spec/helpers/marble-testing.ts

```diff
@@ -1,45 +1,49 @@
-import {Observable} from '../../dist/cjs/Observable';
-import {SubscriptionLog} from '../../dist/cjs/testing/SubscriptionLog';
-import {ColdObservable} from '../../dist/cjs/testing/ColdObservable';
-import {HotObservable} from '../../dist/cjs/testing/HotObservable';
-import {TestScheduler, observableToBeFn, subscriptionLogsToBeFn} from '../../dist/cjs/testing/TestScheduler';
-
-declare const global: any;
-
-export const rxTestScheduler: TestScheduler = global.rxTestScheduler;
-
-export function hot(marbles: string, values?: any, error?: any): HotObservable<any> {
-  if (!global.rxTestScheduler) {
-    throw 'tried to use hot() in async test';
-  }
-  return global.rxTestScheduler.createHotObservable.apply(global.rxTestScheduler, arguments);
-}
-
-export function cold(marbles: string, values?: any, error?: any): ColdObservable<any> {
-  if (!global.rxTestScheduler) {
-    throw 'tried to use cold() in async test';
-  }
-  return global.rxTestScheduler.createColdObservable.apply(global.rxTestScheduler, arguments);
-}
-
-export function expectObservable(observable: Observable<any>,
-                                 unsubscriptionMarbles: string = null): ({ toBe: observableToBeFn }) {
-  if (!global.rxTestScheduler) {
-    throw 'tried to use expectObservable() in async test';
-  }
-  return global.rxTestScheduler.expectObservable.apply(global.rxTestScheduler, arguments);
-}
-
-export function expectSubscriptions(actualSubscriptionLogs: SubscriptionLog[]): ({ toBe: subscriptionLogsToBeFn }) {
-  if (!global.rxTestScheduler) {
-    throw 'tried to use expectSubscriptions() in async test';
-  }
-  return global.rxTestScheduler.expectSubscriptions.apply(global.rxTestScheduler, arguments);
-}
-
-export function time(marbles: string): number {
-  if (!global.rxTestScheduler) {
-    throw 'tried to use time() in async test';
-  }
-  return global.rxTestScheduler.createTime.apply(global.rxTestScheduler, arguments);
+import {Observable} from '../../dist/cjs/Observable';
```

Comment:
There are two new overloads for `hot` and `cold`.

```typescript
export function hot(marbles: string, values?: void, error?: any): HotObservable<string>;
export function hot<V>(marbles: string, values?: { [index: string]: V; }, error?: any): HotObservable<V>;


export function cold(marbles: string, values?: void, error?: any): ColdObservable<string>;
export function cold<V>(marbles: string, values?: { [index: string]: V; }, error?: any): ColdObservable<V>;
```

The first overload, defaults the returned `HotObservable` to be a type of string (ie `ColdObservable<string>`)
The second overload, allows for type inference of the output observable, so that it is typed as a union of the input values.

Given a value  `let values = { a: 1, b: 2, c: 3 }` then the type of `hot('a---b---c---|', values)` will then `HotObservable<number>`;

Given a more complex case (stolen from `mergeAll-spec.ts`)
```typescript
it('should merge two cold Observables at a time with parameter concurrency=2', () => {
    const x = cold(     'a---b---c---|        ');
    const xsubs =     '  ^           !        ';
    const y = cold(        'd---e---f---|     ');
    const ysubs =     '     ^           !     ';
    const z = cold(                 '--g---h-|');
    const zsubs =     '              ^       !';
    const e1 =    hot('--x--y--z--|           ', { x, y, z });
    const e1subs =    '^                     !';
    const expected =  '--a--db--ec--f--g---h-|';

    expectObservable(e1.mergeAll(2)).toBe(expected);
    expectSubscriptions(x.subscriptions).toBe(xsubs);
    expectSubscriptions(y.subscriptions).toBe(ysubs);
    expectSubscriptions(z.subscriptions).toBe(zsubs);
    expectSubscriptions(e1.subscriptions).toBe(e1subs);
  });
```

Given that `x`, `y`, and `z` are all `ColdObservable<string>` the type of `e1` then becomes `HotObservable<ColdObservable<string>>`.  This then gives us proper type information in unit tests and allows removing of several corner cases where we explicitly cast to any to work around the problem.